// proto3記法であることを表している
syntax = "proto3";

package faq;



// FAQ
message FaqComponent {
    uint32 QID =1;
    string scope = 2;
    string service_name = 3;
    string category = 4;
    string question = 5;
    string answer = 6;
    repeated string tag = 7;
}

message faqShowRequest {
    //string timestamp = 1;
}

message showAllRequest {
}

// 検索ワードを受けるリクエスト宣言
// Repeatedにしたほうがいいかは検討の余地あり
message searchWordRequest {
    string word = 1;
}

// QID指定で取りに来た場合のリクエスト宣言
message showQARequest {
    uint32 QID = 1;
}

// 複数FAQを返す際の共通レスポンス
message multiFaqResponse {
    repeated FaqComponent faq = 1;
}

// repeated SOMETHING = SOMETHING[]
message faqShowResponse {
    repeated FaqComponent faq = 1;
}

message showQAResponse {
    FaqComponent faq = 1;
}

// ルーティング
service FaqGateway {
    // 将来的に削除する宣言(showAllに寄せるか、別途目的を明確化した宣言を作る)
    rpc faqShow (faqShowRequest) returns (faqShowResponse) {};
    rpc showAll (showAllRequest) returns (multiFaqResponse) {};
    rpc searchWord (searchWordRequest) returns (multiFaqResponse) {};
    rpc showQA (showQARequest) returns (showQAResponse) {};
}

